scripts/rebase-ledger.js

```diff
@@ -0,0 +1,48 @@
+const fs = require("fs");
+const sc = require("../node_modules/sourcecred/dist/server/api.js").sourcecred;
+const isEqual = require("../node_modules/lodash.isequal");
+
+const rawToLedgerLog = (rawString) =>
+  JSON.parse("[" + rawString.trim().split("\n").join(",") + "]");
+
+fs.readFile("./data/ledger.json", "utf8", (err, data) => {
+  if (err) {
+    console.error(err);
+    return;
+  }
+  const ledgerLog1 = rawToLedgerLog(data);
+  const ledger = sc.ledger.ledger.Ledger.fromEventLog(ledgerLog1);
+  fs.readFile("./data/ledger-temp.json", "utf8", (err, data) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    const ledgerLog2 = rawToLedgerLog(data);
+    const firstIndex = ledgerLog2.findIndex(
+      (x, i) => !isEqual(ledgerLog1[i], x)
+    );
+    if (firstIndex === -1) {
+      throw "No local events to rebase.";
+    }
+    let i = firstIndex;
+    for (; i < ledgerLog2.length; i++) {
+      try {
+        ledger._createAndProcessEvent(ledgerLog2[i].action);
+      } catch {
+        throw "There was a conflict. You will need to manually redo the actions onto the ledger in the main branch.";
+      }
+    }
+    const eventsRebased = i - firstIndex;
+    fs.writeFile("./data/ledger.json", ledger.serialize(), (err) => {
+      if (err) console.log(err);
+      else {
+        console.log(
+          "Success! " +
+            eventsRebased +
+            " local events rebased onto the origin/main ledger."
+        );
+        console.log("Commit and push the changes.");
```

Comment:
Given this is the output (an encouraging console log), wouldn't this script be more aptly named something like:
`check-ledger-conflicts.js <new-data> <old-data>`